apps/ae_backend_service/lib/backend_session.ex

```diff
@@ -290,8 +316,11 @@ defmodule BackendSession do
 
       _ ->
         Logger.error("Other end missusing protocol")
-
         # if timer elapsed, try and force progress, the same goes for if it's been quiet to long....
+        case cancel_timer(fp_timer) do
+          {:ok, time_left} -> Process.send_after(self(), :force_progress, time_left)
+          {:error, _} -> send(__MODULE__, :force_progress)
+        end
```

Comment:
That code is not needed? The timer is set and should fire of when propriate
```suggestion
```

Reply:
So, this code addresses the comment https://github.com/aeternity/ae-channel-service/blob/65857fcfe22e2119fe5147a4f28c06e184ac890d/apps/ae_backend_service/lib/backend_session.ex#L319